MetaMask/metamask-extension
# feat(1463): introduce Dapp Viewed track when user revisits a dapp

## **Description**
Part 2 of https://app.zenhub.com/workspaces/metamask-extension-platform-655e3b56511c30108d50e7b8/issues/gh/metamask/metamask-planning/1463

We want to add Dapp Viewed event for several actions. The next ones one introduced in this PR is emit metrics when 

- refresh a dapp which has already been connected;
- open a tab and input dapp url which has been already connected; 
- navigate to dapp which has already been connected;


Properties are:
```
1. is_first_visit : always false as we have connected this dapp before
2. number_of_accounts - number of accounts in wallet
3. number_of_accounts_connected - number of connected accounts to Dapp
```


## **Related issues**

Fixes: [#1463](https://app.zenhub.com/workspaces/metamask-extension-platform-655e3b56511c30108d50e7b8/issues/gh/metamask/metamask-planning/1463)

## **Manual testing steps**

1. Onboard MM
2. Opt in metrics in `Participate in MetaMetrics`
3. connect to a dapp with 1 or more accounts (eg: [uniswap](https://app.uniswap.org/swap) or [compound)](https://app.compound.finance/?market=usdc-mainnet)
4. refresh the dapp and check events sent in background (search for 'batch')
You would only see `Dapp_viewed ` once after success refresh, comes with properties as
```
is_first_visit: false (always)
number_of_accounts
number_of_accounts_connected
```
5. Or you can open the dapp in a new tab, which triggers event once also, with similar properties above
6. Or you can navigate to the dapp that's connected and opened in a tab before

## **Screenshots/Recordings**



### **Before**



### **After**


https://github.com/MetaMask/metamask-extension/assets/12678455/e237efd7-380a-44b2-bb14-56dfe16fa7af



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
